### ISSUE TITLE
Rate limit exceeded

### ISSUE THREAD

**Issue by PeanutAria** [Spicadox/auto-twitter-space#10]
Just want to let you know that after running it without any problems for almost 6-8 hours, `SLEEP_TIME = 2`, I finally got `error code 88 rate limit exceeded`.

Tried running it again using my main account auth with a 10secs sleep time, unfortunately it only ran for 2 minutes before it got rate limited.

This is the error log from my main. Unfortunately I turned off logging after closing the last ticket, so I don't know when exactly the rate limit started.

```
2023-07-01 23:25:32 [index.py:159] DEBUG    [＊＊＊＊＊] {'errors': [{'code': 88, 'message': 'Rate limit exceeded.'}]}
2023-07-01 23:25:32 [index.py:172] DEBUG    'list' object has no attribute 'get'
Traceback (most recent call last):
  File "B:\!monitoring tools\auto-twitter-space\index.py", line 162, in get_space_tweet_id
    elif rest_id_json.get('errors').get(0).get('code') == 239:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'get'
2023-07-01 23:25:42 [index.py:159] DEBUG    [＊＊＊＊＊＊＊] {'errors': [{'code': 88, 'message': 'Rate limit exceeded.'}]}
2023-07-01 23:25:42 [index.py:172] DEBUG    'list' object has no attribute 'get'
Traceback (most recent call last):
  File "B:\!monitoring tools\auto-twitter-space\index.py", line 162, in get_space_tweet_id
    elif rest_id_json.get('errors').get(0).get('code') == 239:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'list' object has no attribute 'get'
```

UPDATE:
Saw #TwitterDown is trending and a lot of people got the same error whilst browsing Twitter. I guess it had something to do with the error above. I don't know whether to close or keep this issue open. Feel free to close this if you think I should close it.

**Comment by Spicadox** [comment]
Yeah I made a mistake so you're seeing an ugly mess of attribute error being logged,  but the end result is you are being rate-limited. Seems like this new api itself has ~700 request limit(inheriantly) and it's really quite easy to hit it, not to mention taking half a day for rate limit to refresh. I've yet to even hit that measly limit and I've gotten rate-limited simply by possibly sending too many request in a short time span. So to be perfectly honest, I have no idea if I can actually do anything(I'm sure others may have figured something out, but I'm at a lost ¯\_(ツ)_/¯). 


![image](https://github.com/Spicadox/auto-twitter-space/assets/60198162/5b88e441-18c9-4e71-9011-b5fe4a7c9f68)

As of now I'm just going to wait until the storm blows over and as sad as it is to say, I've enjoyed coding this pet project of mine even if it's not as great as others out there, if it's the end of the line, then it is what it is.

TLDR; Don't know if I can even do anything about rate-limit issue and as long as this new api(disregarding the outage and issues going on) is in place and I have no means/knowing of  how to refresh the ~700 request limit, then it might be better to probably not use this script as I'm *hopeful* others will/have figure out a way well before I do.

**Comment by PeanutAria** [comment]
It's okay, I understand. You've done a lot though

> it's not as great as others out there

even if it's true (according to your words) for me the simplicity(?) of it that makes it easier to use as someone like me who has little to zero knowledge of coding. So thank you so much🙏🙇

**Comment by Spicadox** [comment]
Thanks, genuinely really appreciate all the comments and feedbacks! It makes working on this project even if it's for fun, all the more worthwhile. 

For now, it's a waiting game until things are back to working and once that happens, I'll try to continue maintaining this project as initially planned, in hopes of finding a way to make it work 👍
